atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/locks/LockableTest.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.asts.locks;
+
+import static com.palantir.logsafe.testing.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableSet;
+import com.palantir.atlasdb.sweep.asts.locks.Lockable.LockableComparator;
+import com.palantir.lock.v2.LockRequest;
+import com.palantir.lock.v2.LockResponse;
+import com.palantir.lock.v2.LockToken;
+import com.palantir.lock.v2.TimelockService;
+import com.palantir.refreshable.Refreshable;
+import com.palantir.refreshable.SettableRefreshable;
+import java.time.Duration;
+import java.util.Comparator;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.stubbing.OngoingStubbing;
+
+@ExtendWith(MockitoExtension.class)
+public final class LockableTest {
+    private static final TestLockable TEST_LOCKABLE = TestLockable.of(1);
+    private final SettableRefreshable<Duration> lockTimeout = Refreshable.create(Duration.ofSeconds(1));
+    private Lockable<TestLockable> lockable;
+
+    @Mock
+    private TimelockService timelockService;
+
+    @BeforeEach
+    public void setup() {
+        lockable = createLockable(TEST_LOCKABLE);
+    }
+
+    @Test
+    public void comparatorUsesProvidedComparatorForComparison() {
+        TestLockable otherLockable = TestLockable.of(2);
+        Lockable<TestLockable> lockOne = createLockable(TEST_LOCKABLE);
+        Lockable<TestLockable> lockTwo = createLockable(otherLockable);
+        Comparator<TestLockable> comparator = mock(Comparator.class);
+        LockableComparator<TestLockable> lockableComparator = new LockableComparator<>(comparator);
+
+        int _result = lockableComparator.compare(lockOne, lockTwo);
```

Comment:
Is this enforced by some annotation? Generally the convention for these is just to invoke the method call.

Reply:
Huh, I'd have sworn I got a CheckReturnValue error here (otherwise I wouldn't have done it?) - but I cannot reproduce that anymore, so eh, will revert back!